### PR TITLE
Support node_color and docs.show under the config object

### PR DESCRIPTION
**Pull request by b-per**
relates to #279
works in conjunction with https://github.com/dbt-labs/dbt-core/pull/5397

I leveraged @sungchun12's logic and the modifications to dbt Core of https://github.com/dbt-labs/dbt-core/pull/5397 to support defining `docs.show` and `docs.node_color` as node configs and have them update the docs accordingly.

With this approach `docs.show` can be defined in 2 different places. As `docs.show`, as it has been the case in the past versions (and kept like this to avoid breaking changes), and as `config.docs.show` when we define the docs as a config item. 

The big advantage of the config item is that we can configure models at the folder level in `dbt_project.yml`

For now, there is no validation of `node_color` on the `dbt-docs` side but there is some validation on the `dbt-core` side.

The following config under `dbt_project.yml` generates the DAG below:
```
models:
  jaffle_shop:
      materialized: table
      +docs:
        node_color: criMsON
      staging:
        materialized: view
        +docs:
          node_color: "#e76f51"
```
image

### CODE REVIEW

**Comment by sungchun12** [comment]
I know we talked about `+docs` not being intuitive on a surface level for this config, but that's okay given current UX expects a person to do the same with the config: [`persist_docs`](https://docs.getdbt.com/reference/resource-configs/persist_docs)

File: src/app/docs/analysis.html [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <div class="app-links app-sticky">
         <div class="app-title">
             <div class="app-frame app-pad app-flush-bottom">
-                <h1 ng-if="model.docs.show === false">
+                <h1 ng-if="model.docs.show === false || model.config.docs.show === false">
```
Comment (sungchun12) [inline src/app/docs/analysis.html:22]: I like the if else default here

**Comment by sungchun12** [comment]
@b-per can you take a first draft of adding the defensive code: https://gist.github.com/stu-k/aac9d0fb7bf5e325f4fa8d260bd28fc8


based on the comment: https://github.com/dbt-labs/dbt-core/pull/5397#discussion_r902604111

**Comment by b-per** [comment]
I have added Stu's code on color validation and tested it.

With this new code, even thought the `node_color` is set as `"node_color": "invalid_color"`, the docs do not crash anymore and the color is set to the default one. (example below with the model `customers` having an invalid code)
image

The question now is whether we would like to keep the validation on the dbt-docs side only or on both dbt-docs and dbt-core

**Comment by jtcohen6** [comment]
I have a very slight preference for accessing this metadata from node's top-level property _always_ (`node.docs`), rather than reaching into `node.config.docs` — and that we should perform the coordination between those two fields within `dbt-core` (https://github.com/dbt-labs/dbt-core/pull/5397#discussion_r908317055). I appreciate that's a bigger lift.

@dbt-labs/core-language very open to your thoughts here, as the team continues to develop its thinking around what should be a config and what should be a top-level property.

**Comment by sungchun12** [comment]
Doing some manual testing to check for edge cases. No red flags!

I noticed if you import a dbt package that people will have to color code that via code as well. You'll notice `dbt_artifacts` is left alone as expected. 

**What's working well?**

- folder level color configs
- node-specific overrides folder level

image

**Comment by sungchun12** [comment]
TODO: Also, with docs behaving the way that jerco mentioned, we will need to update the dbt-docs code to look under model.docs instead of model.config.docs (not a huge change but just keeping it in mind)

**Comment by sungchun12** [comment]
> 

@jtcohen6 any updates on whether we have the node_color under property vs. config hierarchy?

I have a draft in #288 separated out until we answer the above to then officially merge those changes into this PR. 

This will require changes to https://github.com/dbt-labs/dbt-core/pull/5397 based on the threads there.

I want to make sure we're aligned on approach before we commit code in the wrong design pattern!

**Comment by sungchun12** [comment]
I will own this to ensure the property is at the top-level vs. config level

File: src/app/services/graph.service.js [hunk 2]
```diff
@@ -301,12 +308,18 @@ angular
                 el.data['selected'] = 1;
             }
 
-            if (el.data.docs && el.data.docs.show === false) {
+            // a model can be hidden by the configuration under docs or under config.docs
+            if (! ( _.get(el,['data', 'docs','show'],true)) ) {
                 el.data['hidden'] = 1;
             }
+
+            var color_config = _.get(el, ['data', 'config', 'docs', 'node_color'])
```
Comment (sungchun12) [inline src/app/services/graph.service.js:316]: @b-per Is this still expected? If I remove the 'config' portion in the list, the custom colors do NOT work.

Comment (b-per) [reply src/app/services/graph.service.js:316]: I just did a test with the latest dbt-core and it works without looking at `config`. I think that we said we'd want to read from docs directly so I will push a small change.

**Review by stu-k — APPROVED** [review]
Looks good to me!

**Review by emmyoop — CHANGES_REQUESTED** [review]
Looks good.  Can you please add a changelog entry indicating the change and then it should be ready to merge!

**Comment by b-per** [comment]
I just modified the CHANGELOG 😄

**Review by emmyoop — APPROVED** [review]

**Pull request merged**